sklearn/semi_supervised/_self_training.py

```diff
@@ -237,6 +253,12 @@ def fit(self, X, y):
                 UserWarning,
             )
 
+        if _routing_enabled():
+            routed_params = process_routing(self, "fit", **params)
+        else:
+            routed_params = Bunch()
+            routed_params.base_estimator = Bunch(fit={})
```

Comment:
you need to pre-define these keys, otherwise this line fails with a `NameError`

Isn't this raise in the CI? hmm

Reply:
that "hmm" reminded me of this guy from dark souls
![hmm](https://github.com/scikit-learn/scikit-learn/assets/1663864/6b2f14e7-9e00-4c4d-83ed-a7e11350ccf9)

Reply:
It seems like the pattern works fine? https://github.com/scikit-learn/scikit-learn/blob/65b2571a3d1cd4b0460f3c3ddd5ad46df26f6e4a/sklearn/calibration.py#L373-L375

But I should pass in `fit=params` instead of `fit={}`. Is that what you meant? I also never got the error locally, or on the CI :0